sdk/search/search-documents/src/searchClient.ts

```diff
@@ -230,18 +232,45 @@ export class SearchClient<T> implements IndexDocumentsClient<T> {
    * @param searchText - The search text on which to base autocomplete results.
    * @param suggesterName - The name of the suggester as specified in the suggesters collection that's part of the index definition.
    * @param options - Options to the autocomplete operation.
+   * @example
+   * ```ts
+   * import {
+   *   AzureKeyCredential,
+   *   SearchClient,
+   *   SelectFields,
+   * } from "@azure/search-documents";
+   *
+   * type Model = {
+   *   key: string;
+   *   azure?: { sdk: string | null } | null;
+   * };
+   *
+   * const client = new SearchClient<Model>(
+   *   "endpoint.azure",
+   *   "indexName",
+   *   new AzureKeyCredential("key")
+   * );
+   *
+   * const fields: SelectFields<Model>[] = ["azure/sdk"];
+   *
+   * const autocompleteResult = await client.autocomplete(
+   *   "searchText",
+   *   "suggesterName",
+   *   { searchFields: fields }
+   * );
```

Comment:
I always thought we needed the array of fields to have a strong type, like `["azure/sdk"] as const` rather than assigning it to a weaker type like `SelectFields<Model>[]`. What is the inferred type of `autocompleteResult` here? Is it not refined at all by the fields selected?

Reply:
Results aren't narrowed by the `searchFields` property. All properties are returned unless `select` is specified. `suggest` is an exception, in that it only returns the key by default. In any case, the docs have been updated to accurately reflect how to get the best typing experience - it does involve defining `select` as const.